From 40af308f6514ba9a6ce83e22523558e16c04023c Mon Sep 17 00:00:00 2001
From: Patrick Delaunay <patrick.delaunay@foss.st.com>
Date: Mon, 14 Mar 2022 16:19:47 +0100
Subject: [PATCH 0378/1695] plat-stm32mp2: configure SRAM1 as shadow location

Configure the Secure only access to first 4KB of SRAM1 as shadow
location of OTPs.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7e9c96f4a6561b57782f97c049b1783214d2f05b plat-stm32mp2: configure SRAM1 as shadow location]
Signed-off-by: Patrick Delaunay <patrick.delaunay@foss.st.com>
Change-Id: Ie1d72f9ea31fe59379494c87b60e9cbbae483dad
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/247744
---
 core/arch/arm/plat-stm32mp2/main.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index 5d89f8626..f953d81fa 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -39,6 +39,9 @@ register_phys_mem_pgdir(MEM_AREA_IO_SEC, SAHB_BASE, SAHB_SIZE);
 
 register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE, GIC_SIZE);
 
+/* Map beginning SRAM1 as read write for BSEC shadow */
+register_phys_mem_pgdir(MEM_AREA_RAM_SEC, SRAM1_BASE, SIZE_4K);
+
 #define _ID2STR(id)		(#id)
 #define ID2STR(id)		_ID2STR(id)
 
-- 
2.47.2

